ironfish/src/wallet/wallet.ts

```diff
@@ -1387,6 +1396,20 @@ export class Wallet {
     this.onAccountRemoved.emit(account)
   }
 
+  async removeAccountSynchronous(account: Account, tx?: IDatabaseTransaction): Promise<void> {
+    await this.walletDb.db.withTransaction(tx, async (tx) => {
+      if (account.id === this.defaultAccount) {
+        await this.walletDb.setDefaultAccount(null, tx)
+        this.defaultAccount = null
+      }
+      // no this.walletDb.removeAccount(account, tx) - we don't want to add the account to cleanup list
+      await this.walletDb.removeHead(account, tx)
+      this.accounts.delete(account.id)
+      await this.walletDb.deleteAccount(account, tx)
+      this.onAccountRemoved.emit(account)
```

Comment:
It might be good to move `this.accounts.delete(account.id)` and `this.onAccountRemoved.emit(account)` outside of the database transaction. That way these will only run if the transaction is committed.

It's probably OK in this case because `accounts` is an in-memory cache populated when the node starts.